# Obesity Trend Analysis in India 
## NFHS-5 (2019-20)
### Step 2: Exploratory Data Analysis (EDA)

### Step 1: Importing Libraries

In [ ]:
import pandas as pd
import numpy as np
print("Libraries Imported Successfully")

Libraries imported successfully


### Step 2: Loading the clean dataset

In [3]:
df = pd.read_csv('../data/processed/obesity_cleaned.csv')
print("Data Loaded Succesfully!")

Data loaded succesfully!


### Step 3: Checking Shape

In [4]:
print(f"Rows: {df.shape[0]}")
print(f"Columns: {df.shape[1]}")

Rows: 36
Columns: 17


### Step 4: Preview Dataset

In [5]:
df.head()

,state,zone,overweight_obese_w,overweight_obese_m,education_w,education_m,income,tobacco_w,tobacco_m,alcohol_w,alcohol_m,internet_w,internet_m,blood_sugar_w,blood_sugar_m,bp_w,bp_m
0,Andaman and Nicobar Islands,South,38.1,45.3,52.5,52.3,197278.0,31.3,58.7,5.0,39.1,34.8,46.5,17.5,17.9,25.3,30.2
1,Andhra Pradesh,South,36.3,31.1,39.6,47.9,163746.0,3.8,22.6,0.5,23.3,21.0,48.8,19.5,21.8,25.3,29.0
2,Arunachal Pradesh,Northeast,23.9,27.6,39.4,48.2,190212.0,18.8,50.3,24.2,52.7,52.9,71.6,8.4,11.9,24.9,33.1
3,Assam,Northeast,15.2,16.2,29.6,35.5,90482.0,22.1,51.8,7.3,25.1,28.2,42.3,12.8,16.0,19.1,20.3
4,Bihar,East,15.9,14.7,28.8,42.8,43605.0,5.0,48.8,0.4,15.5,20.6,43.6,12.7,16.2,15.9,18.4


### Step 5: Summary Statistics

In [6]:
df.describe()

,overweight_obese_w,overweight_obese_m,education_w,education_m,income,tobacco_w,tobacco_m,alcohol_w,alcohol_m,internet_w,internet_m,blood_sugar_w,blood_sugar_m,bp_w,bp_m
count,36.000000,36.000000,36.000000,36.000000,33.000000,36.000000,36.000000,36.000000,36.000000,36.000000,36.000000,36.000000,36.000000,36.000000,36.000000
mean,27.680556,27.552778,46.680556,54.436111,173834.090909,12.858333,38.113889,2.908333,24.783333,43.366667,64.927778,13.861111,15.908333,22.477778,26.216667
std,9.820207,9.061677,13.434685,12.900834,91996.226931,14.461483,15.013387,4.904597,11.104967,16.340170,13.488082,4.387598,4.336251,4.426550,6.026774
min,11.500000,13.900000,23.200000,29.400000,43605.000000,0.400000,12.100000,0.200000,0.400000,20.600000,42.100000,6.700000,8.000000,14.900000,15.400000
25%,21.450000,19.550000,34.775000,46.350000,103654.000000,3.125000,26.900000,0.300000,16.950000,30.000000,56.150000,10.150000,13.575000,19.775000,21.625000
50%,28.800000,29.300000,48.550000,54.050000,183333.000000,8.400000,39.850000,0.900000,23.900000,40.650000,66.500000,13.600000,15.650000,22.650000,25.400000
75%,35.050000,33.050000,56.100000,61.450000,212174.000000,17.350000,48.500000,4.475000,32.575000,55.200000,74.975000,17.500000,17.950000,24.925000,30.300000
max,46.200000,45.300000,77.000000,80.900000,431351.000000,61.600000,72.900000,24.200000,52.700000,76.700000,91.900000,24.800000,27.000000,34.500000,41.600000


### Step 6: Zone Wise Distribution

In [11]:
df['zone'].value_counts()

zone
South        8
Northeast    8
North        8
East         4
Central      4
West         4
Name: count, dtype: int64

### Step 7: Zone Wise Average Analysis

In [30]:
zone_obesity = df.groupby('zone')[['overweight_obese_w', 'overweight_obese_m']].mean().round(1)
print(zone_obesity.sort_values('overweight_obese_w', ascending=False))

           overweight_obese_w  overweight_obese_m
zone                                             
South                    36.6                37.2
North                    32.5                31.0
West                     27.2                24.6
Northeast                22.4                25.4
Central                  20.4                19.0
East                     18.4                17.0


### Step 8: Top 5 and Bottom 5 States by Obesity

In [26]:
sorted_w = df[['state','overweight_obese_w']].sort_values('overweight_obese_w', ascending=False)

print("=====Women=====")
print("\nThe top 5 states are:",sorted_w.head())
print("\nThe bottom 5 states are:",sorted_w.tail())

=====Women=====

The top 5 states are:          state  overweight_obese_w
26  Puducherry                46.2
5   Chandigarh                44.0
24   NCT Delhi                41.3
27      Punjab                40.8
30  Tamil Nadu                40.4

The bottom 5 states are:            state  overweight_obese_w
23      Nagaland                14.4
6   Chhattisgarh                14.1
28     Rajasthan                12.9
13     Jharkhand                11.9
21     Meghalaya                11.5


In [28]:
sorted_m = df[['state','overweight_obese_m']].sort_values('overweight_obese_m', ascending=False)

print("=====Men=====")
print("\nThe top 5 states are:",sorted_m.head())
print("\nThe bottom 5 states are:",sorted_m.tail())

=====Men=====

The top 5 states are:                           state  overweight_obese_m
0   Andaman and Nicobar Islands                45.3
26                   Puducherry                43.3
17                  Lakshadweep                41.3
24                    NCT Delhi                38.0
16                       Ladakh                37.8

The bottom 5 states are:            state  overweight_obese_m
13     Jharkhand                15.1
28     Rajasthan                15.0
6   Chhattisgarh                14.9
4          Bihar                14.7
21     Meghalaya                13.9


### Step 9: Correlation Analysis
**Note:** Correlation measures how two variables move together.
- Values close to +1 = strong positive relationship
- Values close to -1 = strong negative relationship
- Values close to 0 = no relationship

### Step 9.1: Correlation - Obesity vs Education

In [34]:
corr_edu = df[['overweight_obese_w', 'overweight_obese_m','education_w', 'education_m']].corr().round(2)
print(corr_edu)

                    overweight_obese_w  overweight_obese_m  education_w  \
overweight_obese_w                1.00                0.88         0.76   
overweight_obese_m                0.88                1.00         0.81   
education_w                       0.76                0.81         1.00   
education_m                       0.68                0.74         0.92   

                    education_m  
overweight_obese_w         0.68  
overweight_obese_m         0.74  
education_w                0.92  
education_m                1.00  


### Step 9.2: Correlation - Obesity vs Income

In [36]:
corr_income = df[['overweight_obese_w','overweight_obese_m','income']].corr().round(2)
print(corr_income)

                    overweight_obese_w  overweight_obese_m  income
overweight_obese_w                1.00                0.88    0.63
overweight_obese_m                0.88                1.00    0.64
income                            0.63                0.64    1.00


### step 9.3: Correlation - Obesity vs Tobacco

In [38]:
corr_tobacco = df[['overweight_obese_w','overweight_obese_m','tobacco_w','tobacco_m']].corr().round(2)
print(corr_tobacco)

                    overweight_obese_w  overweight_obese_m  tobacco_w  \
overweight_obese_w                1.00                0.88      -0.28   
overweight_obese_m                0.88                1.00      -0.08   
tobacco_w                        -0.28               -0.08       1.00   
tobacco_m                        -0.67               -0.47       0.78   

                    tobacco_m  
overweight_obese_w      -0.67  
overweight_obese_m      -0.47  
tobacco_w                0.78  
tobacco_m                1.00  


### Step 9.4: Correlation - Obesity vs Alcohol

In [39]:
corr_alcohol = df[['overweight_obese_w','overweight_obese_m','alcohol_w','alcohol_m']].corr().round(2)
print(corr_alcohol)

                    overweight_obese_w  overweight_obese_m  alcohol_w  \
overweight_obese_w                1.00                0.88      -0.12   
overweight_obese_m                0.88                1.00       0.01   
alcohol_w                        -0.12                0.01       1.00   
alcohol_m                         0.04                0.10       0.69   

                    alcohol_m  
overweight_obese_w       0.04  
overweight_obese_m       0.10  
alcohol_w                0.69  
alcohol_m                1.00  


### Step 9.5: Correlation - Obesity vs Internet

In [41]:
corr_internet = df[['overweight_obese_w', 'overweight_obese_m','internet_w', 'internet_m']].corr().round(2)
print(corr_internet)

                    overweight_obese_w  overweight_obese_m  internet_w  \
overweight_obese_w                1.00                0.88        0.60   
overweight_obese_m                0.88                1.00        0.65   
internet_w                        0.60                0.65        1.00   
internet_m                        0.65                0.61        0.89   

                    internet_m  
overweight_obese_w        0.65  
overweight_obese_m        0.61  
internet_w                0.89  
internet_m                1.00  


### Step 9.6: Correlation - Obesity vs Blood Sugar

In [43]:
corr_sugar = df[['overweight_obese_w', 'overweight_obese_m','blood_sugar_w', 'blood_sugar_m']].corr().round(2)
print(corr_sugar)

                    overweight_obese_w  overweight_obese_m  blood_sugar_w  \
overweight_obese_w                1.00                0.88           0.63   
overweight_obese_m                0.88                1.00           0.49   
blood_sugar_w                     0.63                0.49           1.00   
blood_sugar_m                     0.48                0.36           0.95   

                    blood_sugar_m  
overweight_obese_w           0.48  
overweight_obese_m           0.36  
blood_sugar_w                0.95  
blood_sugar_m                1.00  


### Step 9.7: Correlation - Obesity vs Blood Pressure

In [44]:
corr_bp = df[['overweight_obese_w', 'overweight_obese_m','bp_w', 'bp_m']].corr().round(2)
print(corr_bp)

                    overweight_obese_w  overweight_obese_m  bp_w  bp_m
overweight_obese_w                1.00                0.88  0.61  0.60
overweight_obese_m                0.88                1.00  0.56  0.56
bp_w                              0.61                0.56  1.00  0.88
bp_m                              0.60                0.56  0.88  1.00


### Step 10: Correlation Analysis among Socioeconomic Factors

### Step 10.1: Correlation - Education vs Income

In [46]:
corr_edu_inc = df[['education_w', 'education_m','income']].corr().round(2)
print(corr_edu_inc)

             education_w  education_m  income
education_w         1.00         0.92    0.62
education_m         0.92         1.00    0.56
income              0.62         0.56    1.00


### Step 10.2: Correlation - Education vs Internet

In [47]:
corr_edu_int = df[['education_w', 'education_m','internet_w','internet_m']].corr().round(2)
print(corr_edu_int)

             education_w  education_m  internet_w  internet_m
education_w         1.00         0.92        0.77        0.75
education_m         0.92         1.00        0.69        0.77
internet_w          0.77         0.69        1.00        0.89
internet_m          0.75         0.77        0.89        1.00


### Step 10.3: Correlation - Income vs Internet

In [48]:
corr_inc_int = df[['income','internet_w','internet_m']].corr().round(2)
print(corr_inc_int)

            income  internet_w  internet_m
income        1.00        0.73        0.63
internet_w    0.73        1.00        0.89
internet_m    0.63        0.89        1.00


### Step 10.4: Correlation - Blood Pressure vs Blood Sugar

In [50]:
corr_bp_bs = df[['bp_w','bp_m','blood_sugar_w','blood_sugar_m']].corr().round(2)
print(corr_bp_bs)

               bp_w  bp_m  blood_sugar_w  blood_sugar_m
bp_w           1.00  0.88           0.49           0.45
bp_m           0.88  1.00           0.29           0.28
blood_sugar_w  0.49  0.29           1.00           0.95
blood_sugar_m  0.45  0.28           0.95           1.00
